openbb_platform/CONTRIBUTING.md

```diff
@@ -1,65 +1,179 @@
+2. Community Extensions - built by anyone and primarily maintained by OpenBB (e.g. `openbb-yfinance`)
+3. Independent Extensions - built and maintained independently by anyone
+
+If your extension is of high quality and you think that it would be a good community extension, you can open a PR to the OpenBB Platform repository and we'll review it.
+
+We encourage independent extensions to be shared with the community by publishing them to PyPI.
+
+# How to build OpenBB extensions?
+
+We have a Cookiecutter template that will help you get started. It serves as a jumpstart for your extension development, so you can focus on the data and not on the boilerplate.
 
 Please refer to the [Cookiecutter template](https://github.com/OpenBB-finance/openbb-cookiecutter) and follow the instructions there.
 
-This document will walk you through the steps of adding a new custom extension to the Platform. The high level steps are:
+This document will walk you through the steps of adding a new extension to the OpenBB Platform.
+
+The high level steps are:
 
 - Generate the extension structure
 - Install your dependencies
 - Install your new package
 - Use your extension (either from Python or the API interface)
+- QA your extension
+- Share your extension with the community
 
-## Cookiecuter, a closer look
-
-The Cookiecutter template generates a set of files in which we can find instructions and explanations.
+## Add a custom data source
 
-> Note that the code is functional, so you can just run it and start playing with it.
+You will get your data either from a CSV file, local database or from an API endpoint.
 
-### Get your data
-
-Either from a CSV file, local database or from an API endpoint, you'll need to get your data.
-
-If you don't want to partake in the data standardization framework, you can simply write all the logic straight inside the router file. This is usually the case when you are adding unique data that isn't easily standardizable.
+If you don't want to partake in the data standardization framework, you can simply write all the logic straight inside the router file. This is usually the case when you are adding alternative data that isn't easily standardizable.
 
 Saying that, we recommend following the standardization framework, as it will make your life easier in the long run and unlock a set of features that are only available to standardized data.
 
 When standardizing, all data is defined using two different pydantic models:
 
-1. Define the [query parameters](openbb_platform\platform\provider\openbb_provider\abstract\query_params.py) model.
-2. Define the resulting [data schema](openbb_platform\platform\provider\openbb_provider\abstract\data.py) model.
+1. Define the [query parameters](platform/provider/openbb_provider/abstract/query_params.py) model.
+2. Define the resulting [data schema](platform/provider/openbb_provider/abstract/data.py) model.
 
 > The models can be entirely custom, or inherit from the OpenBB standardized models.
```

Comment:
The models cannot be entirely custom.  What are the specific caveats to the inputs and outputs of any model, standard or not?

Reply:
By entirely custom, I mean that you can define a model and use it without the need to inherit from a standard model. We do this as an example on the cookiecutter.